lighthouse-core/test/scripts/i18n/collect-strings-test.js

```diff
@@ -332,6 +332,56 @@ describe('parseUIStrings', () => {
   });
 });
 
+describe('#_lhlValidityChecks', () => {
+  /* eslint-disable max-len */
+  it('errors when using non-supported custom-formatted ICU format', () => {
+    const message = 'Hello World took {var, badFormat, milliseconds}.';
+    expect(() => collect.convertMessageToCtc(message)).toThrow(
+      /Did not find the expected syntax \(one of 'number', 'date', 'time', 'plural', 'selectordinal', 'select'\) in message "Hello World took {var, badFormat, milliseconds}."$/);
```

Comment:
this was moved from the existing test below. The error message is now less useful, but `intl-messageformat-parser` won't parse this case at all and doesn't give enough information to know exactly what thing threw, so it doesn't seem like a good idea to absorb all `SyntaxError`s just to have the control flow continue to the check that was catching this before.

This is a rare enough occurrence (using a non-existent ICU format) that hopefully the error message context is enough to figure out what's happening without being a total pain.